File: DemoMMOUnity/Assets/Game/Net/Script/Net/CsProtocol/Protocol/Login/StartGame/LoginTapToStartResponse.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using ZJYFrameWork.Collection.Reference;
using ZJYFrameWork.Net.Core;
using ZJYFrameWork.Net.CsProtocol.Buffer;
using ZJYFrameWork.Spring.Utils;

namespace ZJYFrameWork.Net.CsProtocol.Protocol.Login
{
    public class LoginTapToStartResponse : Model, IReference, IPacket
    {
        /// <summary>
        /// 当不能进入游戏时，提示相关信息
        /// </summary>
        public string message { get; set; }

        /// <summary>
        /// 是否能进入游戏
        /// </summary>
        public bool accessGame { get; set; }

        public void Clear()
        {
            message = string.Empty;
            accessGame = default;
        }

        public short ProtocolId()
        {
            return 1014;
        }
    }

    public class LoginTapToStartResponseRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1014;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            if (packet == null)
            {
                return;
            }

            LoginTapToStartResponse data = (LoginTapToStartResponse)packet;
            var message = new ServerMessageWrite(data.ProtocolId(), data);
            var json = JsonConvert.SerializeObject(message);
            buffer.WriteString(json);
        }

        public IPacket Read(ByteBuffer buffer)
        {
            var json = StringUtils.BytesToString(buffer.ToBytes());
            var dict = JsonConvert.DeserializeObject<Dictionary<object, object>>(json);
            var packet = ReferenceCache.Acquire<LoginTapToStartResponse>();
            packet.Clear();
            foreach (var (key, value) in dict)
            {
                var keyString = key.ToString();
                switch (keyString)
                {
                    case "message":
                    {
                        packet.message = value.ToString();
                    }
                        break;
                    case "accessGame":
                    {
                        var valueString = value.ToString();
                        packet.accessGame = bool.Parse(valueString);
                    }
                        break;
                }
            }

            return packet;
        }
    }
}

File: UnityMMOClient/Lua/application/app/net/webSocket/luaProtocol/login/loginTapToStartResponse.lua
---@class LoginTapToStartResponse
local LoginTapToStartResponse = class("LoginTapToStartResponse")
function LoginTapToStartResponse:ctor()
    --- 是否能进入游戏
    ---@type boolean
    self.accessGame = 0
    --- 当不能进入游戏时，提示相关信息
    ---@type string
    self.message = string.empty
end

---@param boolean boolean 是否能进入游戏
---@param message string 当不能进入游戏时，提示相关信息
---@return LoginTapToStartResponse
function LoginTapToStartResponse:new(accessGame, message)
    self.accessGame = accessGame --- boolean
    self.message = message --- java.lang.String
    return self
end

---@return number
function LoginTapToStartResponse:protocolId()
    return 1014
end

---@return string
function LoginTapToStartResponse:write()
    local message = {
        protocolId = self:protocolId(),
        packet = {
            accessGame = self.accessGame,
            message = self.message
        }
    }
    local jsonStr = JSON.encode(message)
    return jsonStr
end

---@return LoginTapToStartResponse
function LoginTapToStartResponse:read(data)

    local packet = self:new(
            data.accessGame,
            data.message)
    return packet
end

--- 是否能进入游戏
---@return boolean 是否能进入游戏
function LoginTapToStartResponse:getAccessGame()
    return self.accessGame
end
--- 当不能进入游戏时，提示相关信息
---@type  string 当不能进入游戏时，提示相关信息
function LoginTapToStartResponse:getMessage()
    return self.message
end



return LoginTapToStartResponse
